### PR TITLE
Add the docbuffer in company-yasnippet

### PR DESCRIPTION
**Pull request by taigacute**
Like this,

image

### CODE REVIEW

File: company-yasnippet.el [hunk 1]
```diff
@@ -97,6 +97,22 @@
        res))
    tables))
 
+(defun company-yasnippet--doc (arg)
+  (let ((template (get-text-property 0 'yas-template arg))
+        (mode (format "%s" major-mode)))
```
Comment (dgutov) [inline company-yasnippet.el]: Why not keep `mode` a symbol value? Then, below you will compare it with `'web-mode` and set to `'html-mode` if true.

Comment (taigacute) [reply company-yasnippet.el]: thanks.I'm not very good at writing elisp.

Comment (dgutov) [reply company-yasnippet.el]: How about:

```
(mode major-mode)
```

File: company-yasnippet.el [hunk 2]
```diff
@@ -97,6 +97,22 @@
...
 
+(defun company-yasnippet--doc (arg)
+  (let ((template (get-text-property 0 'yas-template arg))
+        (mode (symbol-value 'major-mode)))
+    (with-current-buffer (company-doc-buffer)
+      (yas-minor-mode 1)
+      (yas-expand-snippet (yas--template-content template))
+      (delay-mode-hooks
+        (let ((inhibit-message t))
+          (if (string-equal mode 'web-mode)
```
Comment (dgutov) [inline company-yasnippet.el]: Just `(eq` would suffice.

Comment (taigacute) [reply company-yasnippet.el]: ok  i rewrite it .

File: company-yasnippet.el (hunk 1 above)
Comment (taigacute) [reply company-yasnippet.el]: got it! thanks.

**Comment by taigacute** [comment]
A tip for this pr,use these code  wroks lsp ,set  company-backends `(company-capf company-yasnippet)`, Will get a nice complementary experience。
```elisp
  (defun company-yasnippet/disable-after-dot (fun command &optional arg &rest _ignore)
    (if (eq command 'prefix)
        (when-let ((prefix (funcall fun 'prefix)))
          (unless (memq (char-before (- (point) (length prefix))) '(?. ?> ?\())
            prefix))
      (progn
        (when (and arg (not (get-text-property 0 'yas-annotation-patch arg)))
          (let* ((name (get-text-property 0 'yas-annotation arg))
                 (snip (format "%s (Snippet)" name))
                 (len (length arg)))
            (put-text-property 0 len 'yas-annotation snip arg)
            (put-text-property 0 len 'yas-annotation-patch t arg)))
        (funcall fun command arg))))

  (advice-add #'company-yasnippet :around #'company-yasnippet/disable-after-dot)
```

**Comment by dgutov** [comment]
Thanks!

**Comment by dgutov** [comment]
I have a couple of questions regarding your snippet:

* Why `?\(`? Are snippets after an opening paren unlikely to help either?
* Is the `yas-annotation-patch` thing there to clarify the annotation? Is the ` -> ` arrow not enough?

**Comment by dgutov** [comment]
Try this, maybe:

```diff
diff --git a/company-yasnippet.el b/company-yasnippet.el
index e5fded4..f773522 100644
--- a/company-yasnippet.el
+++ b/company-yasnippet.el
@@ -131,7 +131,7 @@ shadow backends that come after it.  Recommended usages:
           (company-grab-symbol)))
     (annotation
      (concat
-      (unless company-tooltip-align-annotations " -> ")
+      " -> "
       (get-text-property 0 'yas-annotation arg)))
     (candidates (company-yasnippet--candidates arg))
     (no-cache t)
```

**Comment by taigacute** [comment]
> Why ?\(? Are snippets after an opening paren unlikely to help either?  
  
..seems like is wrong, maybe it should be removed. I modified it.

> Is the yas-annotation-patch thing there to clarify the annotation? Is the -> arrow not enough?  

config it will look better when use lsp.
image

**Comment by dgutov** [comment]
OK.

I create a var: `company-yasnippet-annotation-fn`. You can change this behavior more easily through it.

**Pull request merged**